storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -59,10 +65,11 @@ type encoder struct {
 
 func newCommitLogSource(opts Options, inspection fs.Inspection) bootstrap.Source {
 	return &commitLogSource{
-		opts:          opts,
-		inspection:    inspection,
-		log:           opts.ResultOptions().InstrumentOptions().Logger(),
-		newIteratorFn: commitlog.NewIterator,
+		opts:            opts,
+		inspection:      inspection,
+		log:             opts.ResultOptions().InstrumentOptions().Logger(),
+		newIteratorFn:   commitlog.NewIterator,
+		cachedShardData: map[string][]shardData{},
```

Comment:
nit: Perhaps it might be nice to wrap up the `[]shardData` in an encapsulating type?  i.e. `map[string]*cachedShardData` just so you can encapsulate other stuff in there and you don't have to do the weird append thing (i.e. you can append to a field in that type rather than append to the `mapName[key]`)

Reply:
done